web/src/main/webapp/app/app.js

```diff
@@ -410,7 +427,14 @@
 
       $window.onbeforeunload = function(event) {
         if (user && user.origin === 'DJUPINTEGRATION' && $stateParams && !$stateParams.signServiceSubmit) {
-          $.get('/api/anvandare/logout');
+          if ($window.sessionStorage.getItem('launchId')) {
```

Comment:
See if it is possible to extract the logic to separate method(s) and reuse them in the three places that you add the header.